base/client.jl

```diff
@@ -71,6 +71,18 @@ function repl_cmd(cmd, out)
         ENV["OLDPWD"] = new_oldpwd
         println(out, pwd())
     else
+        @windows_only begin
+            # build correctly quoted string for cmd.exe
+            special = r".*[ &()[\]{}\^=;!'+,`~]+.*"
+            cmdstr, sep = "", ""
+            for s in cmd.exec
+                dq = ismatch(special, s) ? "\"" : ""
+                cmdstr = cmdstr * sep * dq * s * dq
```

Comment:
It would probably be better to do this through an IOBuffer rather than repeated string concatenation.